### PR TITLE
Add loan approval prediction project

### LINKED ISSUE #1491: Intelligent Loan Approval Prediction System Using Machine Learning
Currently, loan verification is performed manually, which is time-consuming and may lead to inconsistent decisions. There is no intelligent system available to automatically analyze applicant data and predict whether a loan should be approved or rejected. This can result in delays, human errors, and inaccurate loan approval decisions.

I would like to develop an Intelligent Loan Approval Prediction System using Machine Learning. The system will:

preprocess and clean loan applicant data
perform Exploratory Data Analysis (EDA)
build a loan approval prediction model
compare multiple ML algorithms
visualize insights using graphs and charts
help automate the loan approval decision-making process

Alternative approaches include:

manual verification by loan officers
rule-based approval systems using fixed conditions
spreadsheet-based analysis without predictive modeling

However, these approaches are less efficient, less scalable, and may not provide accurate predictions compared to Machine Learning models.

Tech Stack:

Python
Pandas
Scikit-learn
Matplotlib/Seaborn
Jupyter Notebook

This project will help improve decision-making accuracy, reduce manual effort, and provide faster loan approval analysis.

I would like to work on this issue under GSSoC 2026.
Please assign this issue to me.

### PR DESCRIPTION
**Pull request by anu-006**
closes #1491 

## Description

This PR adds a Machine Learning project for predicting whether a loan application will be approved or rejected.

## What was added

- `Loan_Approval_Prediction.ipynb`
  - Exploratory Data Analysis
  - Missing-value handling
  - Feature engineering
  - Categorical feature encoding
  - Train-test split
  - Data preprocessing
  - SMOTE for handling class imbalance
  - Multiple classification models
  - Model evaluation
  - Random Forest hyperparameter optimization using Optuna
  - Final optimized Random Forest model

- `README.md`
  - Project overview
  - Dataset and feature explanation
  - ML workflow
  - Model comparison
  - Setup and execution instructions
  - Beginner-friendly explanations

- `requirements.txt`
  - Required Python dependencies

## Models Used

- Logistic Regression
- Random Forest
- Gradient Boosting
- XGBoost
- Decision Tree
- Optimized Random Forest

## Result

The optimized Random Forest model achieved approximately **80.49% test accuracy**.

### CODE REVIEW

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
Image

can u add this under similar project category @anu-006 
like the way you did earlier @anu-006

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
can u also include the link of the dataset in the  readme also

File: Intelligent_Loan_Approval_Prediction/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+---
+
+### 📦 `requirements.txt`
+
+```text
```
Comment (Sibam-Paul) [inline Intelligent_Loan_Approval_Prediction/requirements.txt:6]: why this is here

File: Intelligent_Loan_Approval_Prediction/README.md
Comment (Sibam-Paul) [inline Intelligent_Loan_Approval_Prediction/README.md:1]: dataset link here

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@anu-006 make the changes

File: Intelligent_Loan_Approval_Prediction/README.md
Comment (anu-006) [reply Intelligent_Loan_Approval_Prediction/README.md:1]: @Sibam-Paul added dataset link in readme file

**Review by Sibam-Paul — APPROVED** [review]

**Comment by Sibam-Paul** [comment]
same project no need make changes to the existing
